### DISCUSSION TITLE
Have the rate limits changed?

### DISCUSSION THREAD

**Discussion by Taegost** [Sellbrite/Sellbrite-API#22] [Q&A]
On [this page](https://developer.sellbrite.com/reference/rate-limits) is states the rate limits are:

```
Clients who use the Sellbrite API are subjected to two rate limits:

The maximum number of API calls per minute is 40 API requests.
The maximum number of API calls per second is 2 API requests.
```

I'm creating some new automation in n8n and I'm floundering at the very beginning of it. It fails anywhere from 2-7 requests in on the `https://api.sellbrite.com/v1/products` endpoint, where I'm feeding it one SKU at a time to pull data as a PUT to update the store_product_url field. 
It does the update, but it keeps giving me the `403 - "Rate Limit Exceeded"` error. 

I have my automation set to pause 2 seconds between each failed attempt, and it only sends a maximum of 20 items every 35 seconds. It works fine if I pass in one or two things at a time, but I have 12 on my list and it always fails between 2-7 attempts in, even after waiting 15 minutes between runs.

I have verified that every step of the automation works fine on it's own, and the entire string works without any problems if I only give it 1 or 2 items at a time.

A couple weeks ago, I was running these exact same endpoints in batches of several hundred without any problems.

I played around with it a bit and did get the entire thing to run by limiting the batches to 5 requests within 12 seconds (Which would be a total of 60 per minute). After I finish the actual work this was holding up, I'll play around with it some more to dial it in, but I suspect that the API rate limits have changed and the documentation was not updated.

**Comment by Taegost** [comment]
tl;dr: It appears the limit is actually around 15 queries per second and 200 per minute, but the API doesn't handle it consistently when you try to send them that quickly. 1 every 300ms seems to be the sweet spot.

I've been playing around with this for a while all day, and I can't track with 100% accuracy, but it appears that we can do a max of 15 queries per second and 200 per minute, but the API is very finicky when we approach either of those limits with larger batches.
It seems like the most consistent is smaller batches sent more quickly, I've had good luck so far throttling it to 1 query every 300ms (which is theoretically 200 requests per minute).
With my full automation running at that rate, it finishes in about 15m40s after sending 1800 queries (900 GET and 900 PUT), which works out to be a real world throughput of ~1.9 queries per second.
Ironically, setting the throttling to 2 queries every 600ms, while the exact same (theoretically), it always fails during the first 200 queries.
